app/controllers/api/v1/hotels_controller.rb

```diff
@@ -4,22 +4,41 @@ class Api::V1::HotelsController < ApplicationController
 
   # GET /api/v1/hotels
   def index
-    @hotels = Hotel.all
+    # @hotels = Hotel.all.with_attached_images
+    @hotels = []
+    # render json: @hotels
+    Hotel.all.each do |hotel|
+      @hotels.push(hotel.as_json.merge({ image: url_for(hotel.image) }))
+    end
 
-    render json: @hotels
+    if @hotels.size.positive?
+      render json: @hotels
+    else
+      render json: { errors: 'Hotels not found' }, status: :not_found
+    end
```

Comment:
@Tadessejemal1 well done on implementing these changes! You killed it with this feature 🥇 🔥 .